cmd/jaeger/internal/extension/jaegerquery/config.go

```diff
@@ -10,20 +10,31 @@ import (
 	"go.opentelemetry.io/collector/config/confighttp"
 
 	queryApp "github.com/jaegertracing/jaeger/cmd/query/app"
-	"github.com/jaegertracing/jaeger/pkg/tenancy"
 )
 
 var _ component.ConfigValidator = (*Config)(nil)
 
 // Config represents the configuration for jaeger-query,
 type Config struct {
 	queryApp.QueryOptionsBase `mapstructure:",squash"`
-	TraceStoragePrimary       string                  `valid:"required" mapstructure:"trace_storage"`
-	TraceStorageArchive       string                  `valid:"optional" mapstructure:"trace_storage_archive"`
-	MetricStorage             string                  `valid:"optional" mapstructure:"metric_storage"`
-	HTTP                      confighttp.ServerConfig `mapstructure:",squash"`
-	GRPC                      configgrpc.ServerConfig `mapstructure:",squash"`
-	Tenancy                   tenancy.Options         `mapstructure:"multi_tenancy"`
+	Connection                Connection `mapstructure:"connection"`
+	Storage                   Storage    `mapstructure:"storage"`
+}
+
+type Connection struct {
```

Comment:
I am not completely sold on this - it's a different pattern from what most OTEL receivers are using, so why do we want this extra level of indirection?

Reply:
@yurishkuro When the `tenancy` field was still in this config, I thought grouping by connection made sense but it's not really needed anymore. I moved `HTTP` and `GRPC` to `Config`.